metaflow/plugins/aws/awslambda/lambda_decorator.py

```diff
@@ -0,0 +1,213 @@
+                                )
+                            )
+                        else:
+                            if float(v) > LAMBDA_MAX_RESOURCES[k]:
+                                raise LambdaException(
+                                    "{r} cannot exceed {max_v} for *@awslambda* (currently set to {v})".format(
+                                        r=k, max_v=LAMBDA_MAX_RESOURCES[k], v=v
+                                    )
+                                )
+                        self.resource_attributes[k] = str(v)
+        self.run_time_limit = get_run_time_limit_for_task(
+            decos, default_limit=DEFAULT_TIMEOUT_SECONDS
+        )
+        if self.run_time_limit < 10:
+            raise LambdaException(
+                "The timeout for step *{step}* should be at "
+                "least 10 seconds for execution on AWS Lambda".format(step=step)
+            )
+        if self.run_time_limit > 900:
+            raise LambdaException(
+                "The timeout for step *{step}* should be no more than "
+                "15 minutes for execution on AWS Lambda".format(step=step)
+            )
+
+    def runtime_step_cli(
+        self, cli_args: "CLIArgs", retry_count: int, max_user_code_retries: int
+    ):
+        # After all attempts to run the user code have failed, we don't need
+        # Lambda anymore. We can execute possible fallback code locally, so we'll
+        # leave args as is
+        if retry_count <= max_user_code_retries:
+            cli_args.commands = ["awslambda", "step"]
+            cli_args.command_args.append(self.package_sha)
+            cli_args.command_args.append(self.package_url)
+            cli_args.command_options.update(self.attributes)
+            cli_args.command_options.update(self.resource_attributes)
+            cli_args.command_options["run-time-limit"] = self.run_time_limit
+            if not R.use_r():
+                cli_args.entrypoint[0] = sys.executable
+
+    @classmethod
+    def _save_package_once(
+        cls, datastore: "MetaflowDataStore", package: "MetaflowPackage"
+    ):
+        if cls.package_url is None:
+            cls.package_url = datastore.save_data(
+                package.sha, TransformableObject(package.blob)
+            )
+            cls.package_sha = package.sha
+
+    def runtime_init(
+        self,
+        flow: "FlowSpec",
+        graph: "FlowGraph",
+        package: "MetaflowPackage",
+        run_id: str,
+    ):
+        if LAMBDA_LOG_GROUP_NAME is None:
+            raise LambdaException(
+                "The *@lambda* decorator requires METAFLOW_LAMBDA_LOG_GROUP_NAME to be "
+                "set to a Cloudwatch log group name"
+            )
+
+        if LAMBDA_ROLE_ARN is None:
+            raise LambdaException(
+                "The *@lambda* decorator requires METAFLOW_LAMBDA_ROLE_ARN to be set to a "
+                "lambda role arn"
+            )
+
+        if LAMBDA_IMAGE_URI is None:
```

Comment:
If no image is specified, we can default to a `python` image.

Reply:
unfortunately you can't use public images with Lambda, so the user would have to set this